File: frontapp/src/quotes.py
import requests
import json

category = 'happiness'
api_url = 'https://api.api-ninjas.com/v1/quotes?category={}'.format(category)
response = requests.get(api_url, headers={'X-Api-Key': 'WObJpDyID75V+qscVkHA+g==bDWyBPUIn1Zo0W7UN'})

# Check the response status
if response.status_code == requests.codes.ok:
    # Parse the response JSON to get the quote
    quote = response.json()[0]['quote']  # Assuming the API returns a list of quotes
    print(f"Quote: {quote}")
else:
    quote = "Your potential is endless. Keep pushing forward!"  # Default quote in case of error

output_file = 'quote.json'
quote_data = {'quote': quote}

with open(output_file, 'w') as file:
    json.dump(quote_data, file)

print(f"Quote exported to {output_file}")

File: frontapp/src/App/quotes.ts
import axios from 'axios';

const category: string = 'happiness'; // Define the category

// Function to fetch a quote
export const fetchQuote = async (): Promise<void> => {
    const apiUrl: string = `https://api.api-ninjas.com/v1/quotes?category=${category}`;

    try {
        const response = await axios.get(apiUrl, {
            headers: {
                'X-Api-Key': 'WObJpDyID75V+qscVkHA+g==bDWyBPUIn1Zo0W7U', // Replace with your actual API key
            },
            // contentType is not needed in axios; it automatically sets it based on the data being sent
        });

        if (response.status === 200 && response.data.length > 0) {
            return response.data[0].quote;
        } else {} // Log the result on success
    } catch (error) {
        if (axios.isAxiosError(error)) {
            // Handle Axios error
            console.error('Error:', error.response?.data || error.message);
        } else {
            // Handle other errors
            console.error('Error:', error);
        }
    }
};

// Call the function to fetch the quote
fetchQuote();
